xmodule/modulestore/xml_importer.py

```
@@ -834,6 +835,51 @@ def import_library_from_xml(*args, **kwargs):
     return list(manager.run_imports())
 
 
+def _process_sequential_prerequisites(block):
+    """
+    Extracts sequential prerequisite information, flags required sequential,
+    and creates a gating relationship.
+
+    Args:
+        block: The sequential block
+    """
+    if not hasattr(block, "xml_attributes") or not block.xml_attributes:
+        log.debug('Block has no xml_attributes property: %s', block)
+        return
+    try:
+        required_content = block.xml_attributes.get('required_content')
+        if not required_content or not isinstance(required_content, str):
+            log.debug('Failed to extract required_content: %s', block.xml_attributes)
+            return
+
+        min_score = block.xml_attributes.get('min_score')
+        min_completion = block.xml_attributes.get('min_completion')
+        if not min_score or not min_completion:
+            log.debug('Failed to extract min_score, and/or min_completion : %s', block.xml_attributes)
+            return
+
+        min_score = int(min_score)
+        min_completion = int(min_completion)
+
+    except (ValueError, TypeError) as e:
+        log.debug('Failed to extract valid required_content, min_score, and/or min_completion : %s', {e})
+        return
+
+    course_key = block.location.course_key
+
+    prerequisite_usage_key = course_key.make_usage_key('sequential', required_content)
+
+    gating_api.add_prerequisite(course_key, prerequisite_usage_key)
+
+    gating_api.set_required_content(
+        course_key,
+        block.location,
+        prerequisite_usage_key,
+        min_score,
+        min_completion
+    )
```

This will raise an exception if the subsection does not exist. It should be okay to mark the import as failed in this scenario (instead of silently ignoring the errors), but we should ensure that the error message is clear to the course author.

---

The API methods `gating_api.add_prerequisite` and `gating_api.set_required_content` don't throw an exception for non-existing subsection. Course author can only see this as a prerequisite with non-existing subsection, so would have to reimport by applying the fixes on the olx/xml (this kind of error is only when course authors are making manual changes on the olx/xml content!).